server/py/framework/db/sqldb/partititioner.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2025 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sqlalchemy
+import sqlalchemy.orm
+import sqlalchemy.sql.compiler
+
+import mlrun.common.db.dialects
+
+
+class RangePartitioner:
+    def __new__(cls, dialect: str):
+        if dialect.startswith(mlrun.common.db.dialects.Dialects.MYSQL):
+            return super().__new__(MySQLRangePartitioner)
+        if dialect.startswith(mlrun.common.db.dialects.Dialects.POSTGRESQL):
+            return super().__new__(PostgresRangePartitioner)
+        raise ValueError(dialect)
+
+    def bootstrap(
+        self,
+        session: sqlalchemy.orm.Session,
+        table_name: str,
+        partition_expression: str,
+        first_partition_name: str,
+        first_partition_upper_bound: str,
+    ):
+        raise NotImplementedError
+
+
+class MySQLRangePartitioner(RangePartitioner):
```

Comment:
```suggestion
class RangePartitionerMySQL(RangePartitioner):
```

 improves clarity by grouping related classes under a common base name, making them easier to find and understand. It follows widely accepted naming conventions like BaseClassSmth, aiding consistency.
 
 
 
 Same req psql